osquery/tables/system/windows/hvci_status.cpp

```diff
@@ -0,0 +1,87 @@
+#include <osquery/sql.h>
+#include <osquery/system.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/core/windows/wmi.h"
+
+namespace osquery {
+namespace tables {
+
+QueryData genHVCIStatus(QueryContext& context) {
+  Row r;
+  QueryData results;
+
+  const WmiRequest wmiSystemReq("SELECT * FROM Win32_DeviceGuard",
+                                (BSTR)L"ROOT\\MICROSOFT\\WINDOWS\\DEVICEGUARD");
+  const std::vector<WmiResultItem>& wmiResults = wmiSystemReq.results();
+  if (wmiResults.empty()) {
+    LOG(WARNING) << "Error retreiving information from WMI.";
+    return results;
+  }
+  for (const auto& data : wmiResults) {
+    long vbsmethod;
+    long codepolicystatusmethod;
+    long umcipolicymethod;
+
+    data.GetString("Version", r["version"]);
+    data.GetString("InstanceIdentifier", r["instance_identifier"]);
+    data.GetLong("VirtualizationBasedSecurityStatus", vbsmethod);
+    data.GetLong("CodeIntegrityPolicyEnforcementStatus",
+                 codepolicystatusmethod);
+    data.GetLong("UsermodeCodeIntegrityPolicyEnforcementStatus",
+                 umcipolicymethod);
+
+    std::string vbsmethod_str;
+    std::map<long, std::string> vbsmethods;
```

Comment:
Why does this need to be a `std::map`? Perhaps this could just be a statically defined array or vector of `std::string`, so:
```
std::vector<std::string> vbs_methods {
    "VBS_NOT_ENABLED",
    "VBS_ENABLED_AND_NOT_RUNNING",
    "VBS_ENABLED_AND_RUNNING",
};
```

Reply:
Let's also move this up _outside_ of the `for` loop, so that we're not redefining it every pass.

Reply:
I was mapping this because the WMI output returned a number, that correlated to a particular setting status (0 = this, 1 = that), but they seem to be in order from 0 to n, so this seems far simpler.

Reply:
Fixed.